### PR TITLE
fix(generator): Page generators accept default options too

### PR DESCRIPTION
**Pull request by dac09**
## What is it?
Fixes typescript generator for pages.

### CODE REVIEW

File: packages/cli/src/commands/generate/page/page.js [hunk 1]
```diff
@@ -100,47 +103,16 @@ export const routes = ({ name, path }) => {
...
-      )}`
-    )
-}
+// @NOTE: Not exporting handler from function
+// As pages need a special handler
+export const {
+  command,
+  description,
+  builder,
+} = createYargsForComponentGeneration({
```
Comment (dthyresson) [inline packages/cli/src/commands/generate/page/page.js:112]: @dac09 Will this still `Generate storybook files`?

**Review by dthyresson — COMMENTED** [review]
Had one question about storybook - but also what is the best way to test?

Pull the branch and `babel node CRWA`?

File: packages/cli/src/commands/generate/page/page.js (hunk 1 above)
Comment (dac09) [reply packages/cli/src/commands/generate/page/page.js:120]: It does @dthyresson - it does yeah. This code does read a little convoluted, renamed a few variables

**Comment by dthyresson** [comment]
Oh @dac09 I meant to ask if tests and stories were generated by default now -- not just optional (but perhaps that's the PR's intent, too?)

**Comment by dac09** [comment]
> Oh @dac09 I meant to ask if tests and stories were generated by default now -- not just optional (but perhaps that's the PR's intent, too?)

I haven't really changed that behaviour here - they are default!

**Comment by dac09** [comment]
> Had one question about storybook - but also what is the best way to test?

`rwt link` is your friend!! If you have issues linking run `yarn policies set-version 1.18.0` :)

**Comment by dthyresson** [comment]
@dac09 approved to merge

**Comment by zakmandhro** [comment]
Hi @dac09, @dthyresson: I just tried the main branch (0.31.2) and the page generator templates are still .js instead of .tsx. Does this still need fixing? If so, I can do a PR.

**Comment by dac09** [comment]
> the page generator template source files are still .js instead of .tsx.
@zakmandhro yep happy for you to take a stab at this. While the template is JS, I don't think the TS version is any different (apart from file extension?) 

I think I missed the case where you supply the param argument to the generator. What I would do probably... is to use the `AvailableRoutes` type thats generated on dev (in your project under `.redwood/types.routes.d.ts`) but can just be imported like this `import {AvailableRoutes} from '@redwoodjs/router`

> Should we pick a convention and make it consistent?
This is a good point, open source is difficult :). Do you have a suggestion for which convention is best? 


Thanks a bunch for offering to help!

**Comment by Tobbe** [comment]
> Also, we now have generators returning inconsistent conventions.
> 
> * Page generator uses plain functions (and uses any type for params!) and returns JSX using the return statement
> * Component and layout generator uses React.FunctionalCompoent (would be nicer if we just used :FC)
> * Cell  generator returns JSX wrapped in parenthesis
> 
> Should we pick a convention and make it consistent?

Personally I do it like this whenever I can
```
export const MessagesOverview: React.FC = () => (
  <div>
    <div className="row">
      <div className="col-md-8 d-md-block">
        <MessageBox />
      </div>
      <div className="col-md-4 d-md-block">
        <MyMessagesList />
      </div>
    </div>
    <WaitingMessages />
    <LatestAnswersMessage />
  </div>
)
```

But I also understand why we'd generate components using this format

```
export const MessagesOverview: React.FC = () => {
  return (
    <div>
      <div className="row">
        <div className="col-md-8 d-md-block">
          <MessageBox />
        </div>
        <div className="col-md-4 d-md-block">
          <MyMessagesList />
        </div>
      </div>
      <WaitingMessages />
      <LatestAnswersMessage />
    </div>
  )
}
```

Because more often than not you'll end up adding a `const [foo, setFoo] = useState()` or something similar, and that's much easier/faster if your component is already generated using the latter format.

**Pull request merged**